rules/list-item.js

```diff
@@ -250,7 +250,7 @@ function validateListItemPrefixCasing(prefix, file) {
 		return false;
 	}
 
-	if (!listItemPrefixCaseAllowList.has(caseOf(firstWord)) && !/\d/.test(firstWord) && !/^["“'(]/.test(firstWord) && !identifierAllowList.has(firstWord)) {
+	if (!listItemPrefixCaseAllowList.has(caseOf(firstWord.replace(/['·]/,''))) && !/\d/.test(firstWord) && !/^["“'(]/.test(firstWord) && !identifierAllowList.has(firstWord)) {
```

Comment:
I think it would be more correct to strip anything non-ASCII. For example, there are many more kinds of apostrophe's.